cli/src/semgrep/target_manager.py

```diff
@@ -523,7 +528,12 @@ def executes_with_shebang(self, path: Path, shebangs: Collection[Shebang]) -> bo
 
     @lru_cache(maxsize=100_000)  # size aims to be 100x of fully caching this repo
     def get_shebang_line(self, path: Path) -> Optional[str]:
-        if not os.access(str(path), os.X_OK | os.R_OK):
+        if not os.path.exists(str(path)):
+            return None
+        stat_results = os.stat(str(path))
+        mode = bin(stat_results.st_mode)
+        permission_bits = mode[len(mode) - 9 :]
```

Comment:
```suggestion
        permission_bits = mode[-9:]
```

`-9` should already refer to nine characters before the end

Reply:
since there are multiple suggestions in both places, would you mind moving this code to `semgrep.util` so we can make fixes all in one place?